lib/ansiblelint/rules/TaskHasNameRule.py

```diff
@@ -31,5 +31,7 @@ class TaskHasNameRule(AnsibleLintRule):
     _nameless_tasks = ['meta', 'debug']
 
     def matchtask(self, file, task):
-        return task.get('name', '') == '' and \
```

Comment:
any reason we can't just change this to

```
    return (task.get('name') and 
            task["action"]["__ansible_module__"] not in self._nameless_tasks)
```

Reply:
In my opinion it is much easier to read the code if we check each condition in a separate step.
By combining the different conditions with and/or operators it becomes complicated to understand the logic.

Reply:
Sorry, but the step through with a missing condition failing through and implicitly returning None is much less readable in my opinion. I'd agree with you if it were a complicated logical expression but it's effectively `return A && !B`